### PR TITLE
Prettier outline buffer, deriving from outline mode

### PR DESCRIPTION
**Pull request by c1-g**
And also added some documentation for all the defcustoms.
Here's the before-and-after
https://pasteboard.co/KgnYk1q.png

### CODE REVIEW

**Comment by manateelazycat** [comment]
@luhuaei Can you help review this patch ?

File: eaf-pdf-viewer.el [hunk 1]
```diff
@@ -167,55 +155,158 @@
     ("o" . "eaf-pdf-outline")
     ("T" . "toggle_trim_white_margin"))
   "The keybinding of EAF PDF Viewer."
-  :type 'cons)
+  :type '(alist :key-type (string :tag "Key bindings (e.g. \"C-n\", \"<f4>\", etc.)")
+                :value-type (string :tag "Python function"))
```
Comment (manateelazycat) [inline eaf-pdf-viewer.el:159]: Not just Python function, include JavaScript function and Elisp function.

**Review by manateelazycat — CHANGES_REQUESTED** [review]

File: eaf-pdf-viewer.el [hunk 2]
```diff
@@ -56,69 +56,57 @@
...
 (defcustom eaf-pdf-dark-mode "follow"
-  ""
-  :type 'string)
+  "Whether to enable inverted color rendering when starting the pdf-viewer app.
+
+Possible values are
+   - \"follow\" Follow the background color of the theme of user.
+   - \"force\" Force inverted color rendering on start-up.
+   - \"ignore\" Don't do inverted rendering."
+  :type '(choice (string :tag "Force inverted color rendering." "force")
```
Comment (luhuaei) [inline eaf-pdf-viewer.el:88]: Please remove whitespace of line end.

I think , `ignore` is duplicate, if user not setting `eaf-pdf-dark-mode` or setting it to `""`, will default is normal.

File: eaf-pdf-viewer.el [hunk 3]
```diff
@@ -243,9 +334,9 @@ when there is no table of contents for the buffer."
                                                               eaf--buffer-id
                                                               "page_total_number")))))
                     (t
-                     (mapcar #'(lambda (line)
+                     (mapcar (lambda (line)
```
[the remark this answers is not in the document]
Comment (c1-g) [reply eaf-pdf-viewer.el:337]: Well the
``` emacs-lisp
#'(lambda ())
```
and
``` emacs-lisp
(lambda ())
```
are all equivalent according to [(elisp) Anonymous-Functions](https://www.gnu.org/software/emacs/manual/html_node/elisp/Anonymous-Functions.html) but I can change this to fit the coding style if you want.

File: eaf-pdf-viewer.el (hunk 2 above)
Comment (c1-g) [reply eaf-pdf-viewer.el:88]: I don't quite understand what you've said. I thought `"ignore"` meant disabling the dark mode altogether, do you want me to remove that as the "other" choice in customize interface?

File: eaf-pdf-viewer.el [hunk 4]
```diff
@@ -167,55 +155,158 @@
...
+;;
+;;
+;;
+
+;;; TODO
+;;
+;;
+;;
+
+;;; Require
```
[the remark this answers is not in the document]
Comment (c1-g) [reply eaf-pdf-viewer.el:186]: I moved all the defcustoms to the `Customize` section of the code. So the other sections are just there as a result of that.

File: eaf-pdf-viewer.el [hunk 5]
```diff
@@ -56,69 +56,57 @@
...
-
-
-;;; Code:
-
-(defvar eaf-pdf-outline-buffer-name "*eaf pdf outline*"
-  "The name of pdf-outline-buffer.")
-
-(defvar eaf-pdf-outline-window-configuration nil
-  "Save window configure before popup outline buffer.")
+(defgroup eaf-pdf-viewer nil
```
Comment (c1-g) [inline eaf-pdf-viewer.el:60]: Here is where I defined the eaf-pdf-viewer group.

File: eaf-pdf-viewer.el (hunk 1 above)
Comment (c1-g) [reply eaf-pdf-viewer.el]: Ok, I fixed it to just `"Function name"` now

File: eaf-pdf-viewer.el [hunk 6]
```diff
@@ -56,69 +56,57 @@
...
-
-
-;;; Code:
 
-(defvar eaf-pdf-outline-buffer-name "*eaf pdf outline*"
-  "The name of pdf-outline-buffer.")
-
-(defvar eaf-pdf-outline-window-configuration nil
-  "Save window configure before popup outline buffer.")
+(defgroup eaf-pdf-viewer nil
```
Comment (c1-g) [inline eaf-pdf-viewer.el:60]: Here's where I defined the `eaf-pdf-viewer` subgroup. @manateelazycat

File: eaf-pdf-viewer.el [hunk 7]
```diff
@@ -56,69 +56,57 @@
...
+  "The PDF viewer application of Emacs application framework."
+  :group 'eaf)
 
 (defcustom eaf-pdf-extension-list
   '("pdf" "xps" "oxps" "cbz" "epub" "fb2" "fbz")
   "The extension list of pdf application."
-  :type 'cons)
+  :type 'list
+  :group 'eaf-pdf-viewer)
```
[the remark this answers is not in the document]
Comment (c1-g) [reply eaf-pdf-viewer.el:68]: Yes, on the 60th line

File: eaf-pdf-viewer.el (hunk 3 above)
[the remark this answers is not in the document]
Comment (manateelazycat) [reply eaf-pdf-viewer.el:337]: > Well the
> 
> ```emacs-lisp
> #'(lambda ())
> ```
> 
> and
> 
> ```emacs-lisp
> (lambda ())
> ```
> 
> are all equivalent according to [(elisp) Anonymous-Functions](https://www.gnu.org/software/emacs/manual/html_node/elisp/Anonymous-Functions.html) but I can change this to fit the coding style if you want.

#' is better way, please don't change those.

File: eaf-pdf-viewer.el (hunk 2 above)
Comment (luhuaei) [reply eaf-pdf-viewer.el:88]: > Possible values are
>   - \"follow\" Follow the background color of the theme of user.
>   - \"force\" Force inverted color rendering on start-up.
>   - \"ignore\" Don't do inverted rendering."

OK, I understand your meant. 

At first, I looked at this document and thought I could only choose one of the above three.(I misunderstand it)

File: eaf-pdf-viewer.el [hunk 8]
```diff
@@ -167,54 +155,157 @@
...
+(defcustom eaf-pdf-outline-buffer-indent 4
+  "The level of indent in the Outline buffer."
+  :type 'integer
+  :group 'eaf-pdf-viewer)
+
+(defvar-local eaf-pdf-outline-pdf-document nil
+  "The PDF filename or buffer corresponding to this outline
+  buffer.")
+
+(defvar-local eaf-pdf--outline-window-configuration nil
```
Comment (c1-g) [inline eaf-pdf-viewer.el:201]: `eaf-pdf--outline-window-configuration` is now a buffer-local variable.

File: eaf-pdf-viewer.el [hunk 9]
```diff
@@ -167,54 +155,157 @@
...
+    (setq eaf-pdf--outline-window-configuration (current-window-configuration))
 
     ;; Insert outline content.
-    (with-current-buffer (get-buffer-create  eaf-pdf-outline-buffer-name)
+    (with-current-buffer outline-buf
       (setq buffer-read-only nil)
       (erase-buffer)
       (insert toc)
-      (setq toc (mapcar (lambda (line)
+      (setq toc (mapcar #'(lambda (line)
```
Comment (c1-g) [inline eaf-pdf-viewer.el:262]: Changed anonymous functions forms to fit the coding style

**Comment by manateelazycat** [comment]
Looks good, thanks for patch.

**Pull request merged**